src/providers/Tracking/PlantingSiteProvider.tsx

```diff
@@ -0,0 +1,235 @@
+  const reportedPlantsResponse = useAppSelector(selectPlantingSiteReportedPlants(reportedPlantsRequestId));
+
+  const allSitesOption = useMemo(() => {
+    const orgId = isAcceleratorRoute ? acceleratorOrganizationId : selectedOrganization.id;
+    if (activeLocale && orgId) {
+      return {
+        name: strings.ALL_PLANTING_SITES,
+        id: -1,
+        organizationId: orgId,
+        plantingSeasons: [],
+      };
+    }
+  }, [activeLocale, isAcceleratorRoute, acceleratorOrganizationId, selectedOrganization]);
+
+  useEffect(() => {
+    const orgId = isAcceleratorRoute ? acceleratorOrganizationId : selectedOrganization.id;
+    if (orgId) {
+      void dispatch(requestSpecies(orgId));
+      void dispatch(requestPlantingSites(orgId));
+      _setSelectedPlantingSite(undefined);
+    }
+  }, [dispatch, isAcceleratorRoute, acceleratorOrganizationId, selectedOrganization, _setSelectedPlantingSite]);
+
+  const allPlantingSites = useMemo(
+    () => (plantingSitesResults && allSitesOption ? [...plantingSitesResults, allSitesOption] : []),
+    [plantingSitesResults]
+  );
+
+  // Function to select a planting site
+  const setSelectedPlantingSite = useCallback(
+    (plantingSiteId: number) => {
+      const foundSite = allPlantingSites.find((site) => site.id === plantingSiteId);
+      _setSelectedPlantingSite(foundSite);
+    },
+    [allPlantingSites]
+  );
+
+  useEffect(() => {
+    if (plantingSite) {
+      const observationsRequest = dispatch(requestPlantingSiteObservations({ plantingSiteId: plantingSite.id }));
+      const resultsRequest = dispatch(requestPlantingSiteObservationResults({ plantingSiteId: plantingSite.id }));
+      const adHocObservationsRequest = dispatch(
+        requestPlantingSiteAdHocObservations({ plantingSiteId: plantingSite.id })
+      );
+      const adHocResultsRequest = dispatch(
+        requestPlantingSiteAdHocObservationResults({ plantingSiteId: plantingSite.id })
+      );
+      const summariesRequest = dispatch(requestPlantingSiteObservationSummaries(plantingSite.id));
+      const historiesRequest = dispatch(requestListPlantingSiteHistories(plantingSite.id));
+      const reportedPlantsRequest = dispatch(requestPlantingSiteReportedPlants(plantingSite.id));
+      setObservationsRequestId(observationsRequest.requestId);
+      setResultsRequestId(resultsRequest.requestId);
+      setAdHocObservationsRequestId(adHocObservationsRequest.requestId);
+      setAdHocResultsRequestId(adHocResultsRequest.requestId);
+      setSummariesRequestId(summariesRequest.requestId);
+      setHistoriesRequestId(historiesRequest.requestId);
+      setReportedPlantsRequestId(reportedPlantsRequest.requestId);
+    }
+  }, [dispatch, plantingSite]);
+
+  useEffect(() => {
+    if (observationsResponse) {
+      if (observationsResponse.status === 'success') {
+        setObservations(observationsResponse.data ?? []);
+      }
```

Comment:
NIT (can be done later) could simplify all these to `if (<bla>Response?.status === 'success') {...` instead of nested if